snap-core/src/main/java/org/esa/snap/core/util/ProductUtils.java

```diff
@@ -574,74 +503,47 @@ private static void checkCanceled(ProgressMonitor pm) throws IOException {
+        return createGeoBoundaryPaths(rasterDataNode, rect, step > 0 ? step : 1, false);
     }
 
     /**
-     * Converts the geographic boundary of the region within the given product into one, two or three shape objects. If
-     * the product does not intersect the 180 degree meridian, a single general path is returned. Otherwise two or three
-     * shapes are created and returned in the order from west to east.
-     * <p>
-     * This method delegates to {@link #createGeoBoundaryPaths(Product, java.awt.Rectangle, int, boolean) createGeoBoundaryPaths(Product, Rectangle, int, boolean)}
-     * and the additional parameter {@code usePixelCenter} is {@code true}.
-     * <p>
-     * The geographic boundary of the given product are returned as shapes comprising (longitude,latitude) pairs.
-     *
-     * @param product the input product
-     * @param region  the region rectangle in product pixel coordinates, can be null for entire product
-     * @param step    the step given in pixels
-     * @return an array of shape objects
-     *
-     * @throws IllegalArgumentException if product is null or if the product's {@link GeoCoding} is null
-     * @see #createGeoBoundary(Product, java.awt.Rectangle, int)
+     * @deprecated use {@link GeoUtils} instead
      */
+    @Deprecated
     public static GeneralPath[] createGeoBoundaryPaths(Product product, Rectangle region, int step) {
         final boolean usePixelCenter = true;
         return createGeoBoundaryPaths(product, region, step, usePixelCenter);
     }
 
+    /**
+     * @deprecated no replacement
+     */
+    @Deprecated
     public static GeneralPath[] createGeoBoundaryPaths(RasterDataNode rasterDataNode, Rectangle region, int step) {
         final boolean usePixelCenter = false;
         return createGeoBoundaryPaths(rasterDataNode, region, step, usePixelCenter);
     }
 
     /**
-     * Converts the geographic boundary of the region within the given product into one, two or three shape objects. If
-     * the product does not intersect the 180 degree meridian, a single general path is returned. Otherwise two or three
-     * shapes are created and returned in the order from west to east.
-     * <p>
-     * The geographic boundary of the given product are returned as shapes comprising (longitude,latitude) pairs.
-     *
-     * @param product        the input product
-     * @param region         the region rectangle in product pixel coordinates, can be null for entire product
-     * @param step           the step given in pixels
-     * @param usePixelCenter {@code true} if the pixel center should be used to create the pathes
-     * @return an array of shape objects
-     *
-     * @throws IllegalArgumentException if product is null or if the product's {@link GeoCoding} is null
-     * @see #createGeoBoundary(Product, java.awt.Rectangle, int, boolean)
+     * @deprecated use {@link GeoUtils} instead
      */
+    @Deprecated
```

Comment:
Add also 'use {@link GeoUtils} instead'?

Reply:
I completed this by @deprecated no replacement